### ISSUE TITLE
Model validation error causes to lose unsaved form data

### ISSUE THREAD

**Issue by mfendeksilverstripe** [dnadesign/silverstripe-elemental#1250]
### Module version(s) affected

5.2.4

### Description

If a content block uses model validation and a validation error is triggered via Save page action some of the changes made on the content block are lost. This seems to be a regression introduced in CMS 5.

### How to reproduce

* install elemental module
* create a very simple image block (code snippet below)

```php
<?php

use DNADesign\Elemental\Models\ElementContent;
use SilverStripe\AssetAdmin\Forms\UploadField;
use SilverStripe\Assets\Image;

class ImageBlock extends ElementContent
{
    private static string $table_name = 'ImageBlock';

    private static string $singular_name = 'image block';

    private static string $plural_name = 'image blocks';

    private static string $description = 'Text block with an image';

    private static array $has_one = [
        'PrimaryImage' => Image::class,
    ];

    public function getType()
    {
        return $this->i18n_singular_name();
    }

    public function getCMSFields()
    {
        $fields = parent::getCMSFields();
        $fields->removeByName([
            'PrimaryImage',
            'PrimaryImageID',
        ]);
        $fields->addFieldsToTab('Root.Main', [
            UploadField::create('PrimaryImage', 'Primary Image')
        ]);

        return $fields;
    }

    public function validate()
    {
        $validationResult = parent::validate();

        if ($this->PrimaryImageID > 0) {
            $validationResult->addError('This block does not support images at all');
        }

        return $validationResult;
    }
}
```

* create a page with your image block (should look like this)

![Screenshot 2024-09-05 at 8 24 12 AM](https://github.com/user-attachments/assets/cdafe7cb-6645-43d6-984e-5ed95e5167b4)

![Screenshot 2024-09-05 at 8 24 23 AM](https://github.com/user-attachments/assets/49772e30-fb72-4249-bf2e-93d9f021f38b)

* edit text by adding another line of text and choose an existing image to be attached to the block, it should look like this:

![Screenshot 2024-09-05 at 8 27 32 AM](https://github.com/user-attachments/assets/14bb0439-9af9-4989-a464-527d3004321e)

* click Save button on the very bottom (Save whole page action)

We can see the validation message displayed at the top (this is expected).

![Screenshot 2024-09-05 at 8 32 08 AM](https://github.com/user-attachments/assets/c6f215da-9736-483f-99ee-aad328fc7b35)

Expected behaviour:

All unsaved changes are present on the block.

![Screenshot 2024-09-05 at 8 30 49 AM](https://github.com/user-attachments/assets/ed1e9274-265a-410d-805f-57b376c45913)

Actual behaviour:

Attached image change is lost while text edit change is retained.

![Screenshot 2024-09-05 at 8 28 35 AM](https://github.com/user-attachments/assets/761a8565-73c4-4b90-8cf4-5c49b220a1d8)

### Additional Context

Observed that using block save action doesn't have this issue.

![Screenshot 2024-09-05 at 8 34 04 AM](https://github.com/user-attachments/assets/95e9275f-e47d-4451-bcfc-bb02aecd91b7)

Validation error is issued and all unsaved data remains intact.

![Screenshot 2024-09-05 at 8 35 01 AM](https://github.com/user-attachments/assets/8d3860a6-1eea-4093-807e-eb921b5f4cb1)

### Validations

- [X] Check that there isn't already an issue that reports the same bug
- [X] Double check that your reproduction steps work in a fresh installation of [`silverstripe/installer`](https://github.com/silverstripe/silverstripe-installer) (with any code examples you've provided)

**Comment by GuySartorelli** [comment]
I assume this is for inline-editable elemental blocks?
Until CMS 5.3.0 which has not yet been released, validation for inline-editable elemental blocks is simply not feasible and is expected to break things. Do not use validation for inline-editable elemental blocks until CMS 5.3.0.

If this is for non-inline-editable blocks let me know and I'll reopen.

**Comment by mfendeksilverstripe** [comment]
@GuySartorelli  yes, this is for inline-editable blocks only. Good to know that this will be addressed in CMS 5.3. Happy to keep this issue closed.
